### PR TITLE
Restore error for missing default target

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

**Comment by natebosch** [comment]
See #1426

File: build_config/lib/src/parse.dart [hunk 1]
```diff
@@ -124,6 +124,10 @@ BuildConfig parseFromMap(String packageName,
       sources: const InputSet(),
     );
   }
+  if (!buildTargets.containsKey(defaultTarget)) {
```
Comment (kevmoo) [inline build_config/lib/src/parse.dart]: No test?

File: build_config/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+## 0.2.6+2
```
Comment (jakemac53) [inline build_config/CHANGELOG.md]: did one of these already go out?

Comment (natebosch) [reply build_config/CHANGELOG.md]: Yes

File: build_config/lib/src/parse.dart (hunk 1 above)
Comment (natebosch) [reply build_config/lib/src/parse.dart]: Done

**Comment by natebosch** [comment]
Added test. PTAL

**Pull request merged**